deploy.sh

```diff
@@ -78,10 +63,10 @@ function preprocess_and_publish {
   # commit to destination branch with shorthash in message
   git commit -am "update using ${SOURCE_BRANCH}/${shorthash}" || abort "Failed to commit to destination branch '${DEPLOY_BRANCH}'"
 
-  # the "-u" here is if the branch was created
-  git push -u "${GH_REMOTE}" "${DEPLOY_BRANCH}" || abort "Failed to push to '${GH_REMOTE}/${DEPLOY_BRANCH}'"
+  # executes a forced "git subtree push" (git subtree does not take a force so we have to nest)
+  git push "${GH_REMOTE}" `git subtree split --prefix "${WEBSITE_DIR_PATH}" "${SOURCE_BRANCH}"`:"${DEPLOY_BRANCH}" --force || abort "Failed to push to '${GH_REMOTE}/${DEPLOY_BRANCH}'"
   echo "Successfully pushed to ${GH_REMOTE}/${DEPLOY_BRANCH}"
-  git checkout "${SOURCE_BRANCH}" || abort "Failed to checkout source branch after publishing"
+  git reset --hard HEAD~1 || abort "Failed to reset after publishing"
```

Comment:
I don't quite understand this change 🤔

Reply:
Because of
```
  # commit to destination branch with shorthash in message
  git commit -am "update using ${SOURCE_BRANCH}/${shorthash}" || abort "Failed to commit to destination branch '${DEPLOY_BRANCH}'"
```

Reply:
Ah I see; the `git checkout` wasn't doing anything because the source branch was _already_ checked out. I hadn't realized this was being done on a branch.